{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f404ae11",
   "metadata": {},
   "source": [
    "# **1. Getting ready for read alignment**\n",
    "\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9a5c6620",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 6.9G\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson 3.5G Oct 20 14:36 SRR099988_1.Trimmed.fastq.gz\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson 3.5G Oct 20 14:36 SRR099988_2.Trimmed.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh ~/scratch/Project/data/Trimming/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "43f648c6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-10-27 14:55:47--  http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\n",
      "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
      "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 983659424 (938M) [application/x-gzip]\n",
      "Saving to: ‘hg38.fa.gz’\n",
      "\n",
      "hg38.fa.gz          100%[===================>] 938.09M  13.2MB/s    in 50s     \n",
      "\n",
      "2023-10-27 14:56:38 (18.6 MB/s) - ‘hg38.fa.gz’ saved [983659424/983659424]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d17e4e02",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ~/scratch/Project2/\n",
    "!mkdir ~/scratch/Project2/ReadAlignment/\n",
    "!mv hg38.fa.gz ~/scratch/Project2/ReadAlignment/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "aaa946ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "We obtained data from UCSC genome browser's UCSC Golden Path, use Bash's wget command and link \n",
    "(http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz) to get access to the reference genome \n",
    "'''\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e38d4050",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 939M\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson 939M Jan 16  2014 hg38.fa.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh ~/scratch/Project2/ReadAlignment/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0ccaaaa2",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gunzip ~/scratch/SRR099988_1.Trimmed.fastq.gz\n",
    "!gunzip ~/scratch/SRR099988_2.Trimmed.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3637229e",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Our referece genome contains NNNs, lower and upper case nuceotide sequences. \n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9ac4084c",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "Uppercase:non-repeating sequences\n",
    "Lowercase: Repeating sequences (from RepeatMasker and Tandem Repeats Finder).\n",
    "NNNs: unknown/ambiguous sequences\n",
    "'''\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9627b913",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: bwa (alignment via Burrows-Wheeler transformation)\r\n",
      "Version: 0.7.17-r1188\r\n",
      "Contact: Heng Li <lh3@sanger.ac.uk>\r\n",
      "\r\n",
      "Usage:   bwa <command> [options]\r\n",
      "\r\n",
      "Command: index         index sequences in the FASTA format\r\n",
      "         mem           BWA-MEM algorithm\r\n",
      "         fastmap       identify super-maximal exact matches\r\n",
      "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\r\n",
      "         aln           gapped/ungapped alignment\r\n",
      "         samse         generate alignment (single ended)\r\n",
      "         sampe         generate alignment (paired ended)\r\n",
      "         bwasw         BWA-SW for long queries\r\n",
      "\r\n",
      "         shm           manage indices in shared memory\r\n",
      "         fa2pac        convert FASTA to PAC format\r\n",
      "         pac2bwt       generate BWT from PAC\r\n",
      "         pac2bwtgen    alternative algorithm for generating BWT\r\n",
      "         bwtupdate     update .bwt to the new format\r\n",
      "         bwt2sa        generate SA from BWT and Occ\r\n",
      "\r\n",
      "Note: To use BWA, you need to first index the genome with `bwa index'.\r\n",
      "      There are three alignment algorithms in BWA: `mem', `bwasw', and\r\n",
      "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\r\n",
      "      first. Please `man ./bwa.1' for the manual.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "\n",
    "!bwa"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1be29a73",
   "metadata": {},
   "source": [
    "# **2. Read alignment**\n",
    "\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "da6c9fca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Write your commands here. Insert more cells if necessary\n",
    "!gunzip ~/scratch/Project2/ReadAlignment/hg38.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "becbd45d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 3.1G\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson 3.1G Jan 16  2014 hg38.fa\r\n"
     ]
    }
   ],
   "source": [
    "#Run \"ls -lh\" to show what additional files were created in addition to your original fasta files.\n",
    "!ls -lh ~/scratch/Project2/ReadAlignment/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d477ef96",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 25.08 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=6418572210, availableWord=463634060\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 99999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 1099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 120 iterations done. 1199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 130 iterations done. 1299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 140 iterations done. 1399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 150 iterations done. 1499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 160 iterations done. 1599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 170 iterations done. 1699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 180 iterations done. 1799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 190 iterations done. 1899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 200 iterations done. 1999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 210 iterations done. 2099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 220 iterations done. 2199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 230 iterations done. 2299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 240 iterations done. 2399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 250 iterations done. 2499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 260 iterations done. 2599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 270 iterations done. 2699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 280 iterations done. 2799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 290 iterations done. 2899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 300 iterations done. 2999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 310 iterations done. 3099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 320 iterations done. 3199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 330 iterations done. 3299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 340 iterations done. 3399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 350 iterations done. 3499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 360 iterations done. 3599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 370 iterations done. 3699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 380 iterations done. 3799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 390 iterations done. 3899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 400 iterations done. 3999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 410 iterations done. 4099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 420 iterations done. 4199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 430 iterations done. 4299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 440 iterations done. 4399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 450 iterations done. 4499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 460 iterations done. 4599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 470 iterations done. 4699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 480 iterations done. 4799999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 490 iterations done. 4899999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 500 iterations done. 4999999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 510 iterations done. 5099999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 520 iterations done. 5199999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 530 iterations done. 5299999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 540 iterations done. 5399999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 550 iterations done. 5499999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 560 iterations done. 5599999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 570 iterations done. 5699999986 characters processed.\n",
      "[BWTIncConstructFromPacked] 580 iterations done. 5798165394 characters processed.\n",
      "[BWTIncConstructFromPacked] 590 iterations done. 5886412978 characters processed.\n",
      "[BWTIncConstructFromPacked] 600 iterations done. 5964843650 characters processed.\n",
      "[BWTIncConstructFromPacked] 610 iterations done. 6034549010 characters processed.\n",
      "[BWTIncConstructFromPacked] 620 iterations done. 6096499330 characters processed.\n",
      "[BWTIncConstructFromPacked] 630 iterations done. 6151556914 characters processed.\n",
      "[BWTIncConstructFromPacked] 640 iterations done. 6200488210 characters processed.\n",
      "[BWTIncConstructFromPacked] 650 iterations done. 6243974434 characters processed.\n",
      "[BWTIncConstructFromPacked] 660 iterations done. 6282621122 characters processed.\n",
      "[BWTIncConstructFromPacked] 670 iterations done. 6316966322 characters processed.\n",
      "[BWTIncConstructFromPacked] 680 iterations done. 6347488418 characters processed.\n",
      "[BWTIncConstructFromPacked] 690 iterations done. 6374612482 characters processed.\n",
      "[BWTIncConstructFromPacked] 700 iterations done. 6398716386 characters processed.\n",
      "[BWTIncConstructFromPacked] 710 iterations done. 6418572210 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 710 iterations.\n",
      "[bwa_index] 3014.53 seconds elapse.\n",
      "[bwa_index] Update BWT... 16.69 sec\n",
      "[bwa_index] Pack forward-only FASTA... 15.60 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 850.52 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index -a bwtsw /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.fa\n",
      "[main] Real time: 3932.106 sec; CPU: 3922.488 sec\n"
     ]
    }
   ],
   "source": [
    "!bwa index -a bwtsw ~/scratch/Project2/ReadAlignment/hg38.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "217aefa6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 4000814 sequences (240000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 1597989, 15, 16)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 94, 262)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 650)\n",
      "[M::mem_pestat] mean and std.dev: (130.58, 80.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 844)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 151, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 382)\n",
      "[M::mem_pestat] mean and std.dev: (160.09, 64.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 472)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (200, 226, 3091)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8873)\n",
      "[M::mem_pestat] mean and std.dev: (1763.40, 2491.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11764)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 130, 169)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 297)\n",
      "[M::mem_pestat] mean and std.dev: (130.47, 40.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 361)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000814 reads in 319.654 CPU sec, 28.931 real sec\n",
      "[M::process] read 4000830 sequences (240000034 bp)...\n",
      "[M::process] read 4000790 sequences (240000054 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (24, 1598576, 9, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (107, 141, 299)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 683)\n",
      "[M::mem_pestat] mean and std.dev: (157.90, 91.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 875)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 381)\n",
      "[M::mem_pestat] mean and std.dev: (159.59, 64.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 471)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_process_seqs] Processed 4000830 reads in 313.455 CPU sec, 14.429 real sec\n",
      "[M::process] read 4000818 sequences (240000014 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (15, 1599476, 15, 11)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 112, 238)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 512)\n",
      "[M::mem_pestat] mean and std.dev: (147.21, 99.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 649)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (159.02, 64.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (174, 294, 661)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1635)\n",
      "[M::mem_pestat] mean and std.dev: (409.60, 331.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2122)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (96, 146, 313)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 747)\n",
      "[M::mem_pestat] mean and std.dev: (213.91, 185.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 964)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000790 reads in 313.377 CPU sec, 14.465 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 1598554, 12, 11)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (59, 97, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 344)\n",
      "[M::mem_pestat] mean and std.dev: (100.75, 60.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 439)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.61, 64.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (177, 305, 2562)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7332)\n",
      "[M::mem_pestat] mean and std.dev: (933.25, 1144.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 9717)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (105, 124, 151)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (13, 243)\n",
      "[M::mem_pestat] mean and std.dev: (127.09, 35.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 289)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000836 sequences (240000114 bp)...\n",
      "[M::mem_process_seqs] Processed 4000818 reads in 309.489 CPU sec, 14.298 real sec\n",
      "[M::process] read 4000786 sequences (240000068 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (36, 1598778, 16, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (86, 134, 205)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 443)\n",
      "[M::mem_pestat] mean and std.dev: (144.46, 75.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 562)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 375)\n",
      "[M::mem_pestat] mean and std.dev: (157.99, 63.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 463)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (159, 296, 1352)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3738)\n",
      "[M::mem_pestat] mean and std.dev: (318.46, 354.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4931)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000836 reads in 314.288 CPU sec, 15.898 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (31, 1600196, 9, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 101, 157)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 321)\n",
      "[M::mem_pestat] mean and std.dev: (111.90, 57.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 403)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 149, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 374)\n",
      "[M::mem_pestat] mean and std.dev: (157.62, 63.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 462)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 141, 190)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 366)\n",
      "[M::mem_pestat] mean and std.dev: (124.82, 37.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 454)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4000786 reads in 313.415 CPU sec, 15.815 real sec\n",
      "[M::process] read 4000820 sequences (240000008 bp)...\n",
      "[M::process] read 4000858 sequences (240000030 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 1600040, 21, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (69, 109, 190)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 432)\n",
      "[M::mem_pestat] mean and std.dev: (130.54, 79.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 553)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 147, 196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 368)\n",
      "[M::mem_pestat] mean and std.dev: (156.31, 62.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 454)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (184, 227, 397)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 823)\n",
      "[M::mem_pestat] mean and std.dev: (240.28, 153.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1036)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000820 reads in 309.983 CPU sec, 15.239 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (44, 1599795, 18, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (74, 127, 173)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 371)\n",
      "[M::mem_pestat] mean and std.dev: (133.20, 72.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 470)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 147, 196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 368)\n",
      "[M::mem_pestat] mean and std.dev: (156.33, 62.47)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 454)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (178, 228, 464)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1036)\n",
      "[M::mem_pestat] mean and std.dev: (302.89, 200.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1322)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 117, 224)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 492)\n",
      "[M::mem_pestat] mean and std.dev: (126.30, 51.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 626)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000822 sequences (240000104 bp)...\n",
      "[M::mem_process_seqs] Processed 4000858 reads in 315.922 CPU sec, 15.490 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (9, 1596781, 8, 8)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 151, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 379)\n",
      "[M::mem_pestat] mean and std.dev: (159.91, 64.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 468)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::process] read 4000838 sequences (240000090 bp)...\n",
      "[M::mem_process_seqs] Processed 4000822 reads in 317.462 CPU sec, 14.275 real sec\n",
      "[M::process] read 4000838 sequences (240000104 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 1597334, 19, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (79, 109, 144)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 274)\n",
      "[M::mem_pestat] mean and std.dev: (105.00, 46.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 339)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 381)\n",
      "[M::mem_pestat] mean and std.dev: (159.50, 64.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 471)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (204, 258, 1108)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2916)\n",
      "[M::mem_pestat] mean and std.dev: (472.81, 525.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3820)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000838 reads in 317.020 CPU sec, 16.248 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (19, 1597700, 13, 16)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 139, 229)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 551)\n",
      "[M::mem_pestat] mean and std.dev: (129.00, 70.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 712)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.99, 64.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (144, 287, 881)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2355)\n",
      "[M::mem_pestat] mean and std.dev: (438.00, 376.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3092)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (84, 130, 175)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 357)\n",
      "[M::mem_pestat] mean and std.dev: (114.57, 40.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 448)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000786 sequences (240000046 bp)...\n",
      "[M::mem_process_seqs] Processed 4000838 reads in 315.999 CPU sec, 15.916 real sec\n",
      "[M::process] read 4000824 sequences (240000034 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (25, 1598551, 17, 10)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (109, 142, 220)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 442)\n",
      "[M::mem_pestat] mean and std.dev: (165.33, 81.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 553)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.56, 64.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (254, 922, 3595)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 10277)\n",
      "[M::mem_pestat] mean and std.dev: (1891.29, 1899.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 13618)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (98, 140, 322)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 770)\n",
      "[M::mem_pestat] mean and std.dev: (165.78, 109.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 994)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4000786 reads in 316.379 CPU sec, 16.255 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (27, 1598480, 14, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (79, 111, 145)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 277)\n",
      "[M::mem_pestat] mean and std.dev: (112.69, 51.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 343)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 375)\n",
      "[M::mem_pestat] mean and std.dev: (157.99, 63.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 463)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (161, 204, 405)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 893)\n",
      "[M::mem_pestat] mean and std.dev: (234.00, 159.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1137)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::process] read 4000824 sequences (240000096 bp)...\n",
      "[M::mem_process_seqs] Processed 4000824 reads in 307.904 CPU sec, 13.717 real sec\n",
      "[M::process] read 4000808 sequences (240000074 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (29, 1599486, 13, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (90, 119, 225)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 495)\n",
      "[M::mem_pestat] mean and std.dev: (131.38, 70.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 630)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 149, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 377)\n",
      "[M::mem_pestat] mean and std.dev: (157.83, 63.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 466)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (160, 305, 414)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 922)\n",
      "[M::mem_pestat] mean and std.dev: (238.45, 121.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1176)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000824 reads in 313.107 CPU sec, 15.774 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (41, 1599232, 14, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (67, 97, 162)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 352)\n",
      "[M::mem_pestat] mean and std.dev: (104.03, 55.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 447)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 148, 197)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 371)\n",
      "[M::mem_pestat] mean and std.dev: (156.73, 63.01)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 458)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (171, 254, 753)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1917)\n",
      "[M::mem_pestat] mean and std.dev: (276.18, 208.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2499)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 93, 181)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 399)\n",
      "[M::mem_pestat] mean and std.dev: (106.42, 84.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 508)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000826 sequences (240000116 bp)...\n",
      "[M::mem_process_seqs] Processed 4000808 reads in 307.182 CPU sec, 13.733 real sec\n",
      "[M::process] read 4000816 sequences (240000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (33, 1598605, 10, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (66, 93, 144)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 300)\n",
      "[M::mem_pestat] mean and std.dev: (103.91, 55.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 378)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 148, 197)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 371)\n",
      "[M::mem_pestat] mean and std.dev: (157.07, 63.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 458)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 129, 641)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1707)\n",
      "[M::mem_pestat] mean and std.dev: (299.40, 305.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2240)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (106, 124, 244)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 520)\n",
      "[M::mem_pestat] mean and std.dev: (176.25, 111.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 658)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000826 reads in 315.351 CPU sec, 16.764 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (8, 1597456, 13, 6)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 151, 202)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 382)\n",
      "[M::mem_pestat] mean and std.dev: (160.08, 64.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 472)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (141, 201, 1615)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4563)\n",
      "[M::mem_pestat] mean and std.dev: (581.08, 914.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6037)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::process] read 4000802 sequences (240000060 bp)...\n",
      "[M::mem_process_seqs] Processed 4000816 reads in 313.356 CPU sec, 14.004 real sec\n",
      "[M::process] read 4000784 sequences (240000072 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (13, 1598939, 21, 9)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 143, 256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 546)\n",
      "[M::mem_pestat] mean and std.dev: (137.18, 48.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 691)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 150, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 379)\n",
      "[M::mem_pestat] mean and std.dev: (159.54, 64.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 468)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 178, 464)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1146)\n",
      "[M::mem_pestat] mean and std.dev: (220.42, 180.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1487)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4000802 reads in 318.344 CPU sec, 17.072 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 1598920, 14, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (89, 159, 304)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 734)\n",
      "[M::mem_pestat] mean and std.dev: (149.00, 81.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 949)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.98, 64.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 433, 1077)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2937)\n",
      "[M::mem_pestat] mean and std.dev: (345.55, 319.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3867)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000784 reads in 314.207 CPU sec, 13.954 real sec\n",
      "[M::process] read 4000820 sequences (240000078 bp)...\n",
      "[M::process] read 4000820 sequences (240000086 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (17, 1599949, 18, 3)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 115, 154)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 312)\n",
      "[M::mem_pestat] mean and std.dev: (110.38, 39.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 391)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.59, 64.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (149, 302, 2747)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 7943)\n",
      "[M::mem_pestat] mean and std.dev: (1486.94, 2356.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 10912)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000820 reads in 313.162 CPU sec, 16.049 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (22, 1598935, 22, 5)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (63, 101, 146)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 312)\n",
      "[M::mem_pestat] mean and std.dev: (111.36, 58.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 395)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 375)\n",
      "[M::mem_pestat] mean and std.dev: (158.05, 63.77)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 463)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (143, 240, 633)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1613)\n",
      "[M::mem_pestat] mean and std.dev: (356.67, 316.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2103)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000820 reads in 307.118 CPU sec, 13.671 real sec\n",
      "[M::process] read 4000818 sequences (240000020 bp)...\n",
      "[M::process] read 4000808 sequences (240000048 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (29, 1599807, 12, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 105, 180)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 380)\n",
      "[M::mem_pestat] mean and std.dev: (117.63, 56.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 480)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 149, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 374)\n",
      "[M::mem_pestat] mean and std.dev: (157.61, 63.57)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 462)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 368, 549)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1397)\n",
      "[M::mem_pestat] mean and std.dev: (360.75, 308.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1821)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (80, 119, 140)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 260)\n",
      "[M::mem_pestat] mean and std.dev: (109.00, 54.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 327)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000818 reads in 312.151 CPU sec, 15.873 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (40, 1601288, 13, 15)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (89, 121, 187)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 383)\n",
      "[M::mem_pestat] mean and std.dev: (134.28, 68.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 481)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 147, 196)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 368)\n",
      "[M::mem_pestat] mean and std.dev: (156.13, 62.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 454)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (160, 188, 464)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1072)\n",
      "[M::mem_pestat] mean and std.dev: (219.64, 122.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1376)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 152, 175)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (16, 281)\n",
      "[M::mem_pestat] mean and std.dev: (141.15, 38.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 334)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000808 reads in 305.019 CPU sec, 14.333 real sec\n",
      "[M::process] read 4000832 sequences (240000052 bp)...\n",
      "[M::process] read 4000818 sequences (240000040 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (39, 1600333, 13, 10)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 99, 141)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 279)\n",
      "[M::mem_pestat] mean and std.dev: (99.28, 42.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 348)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 148, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 374)\n",
      "[M::mem_pestat] mean and std.dev: (157.29, 63.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 462)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 266, 516)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1398)\n",
      "[M::mem_pestat] mean and std.dev: (234.91, 187.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1839)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (100, 126, 166)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 298)\n",
      "[M::mem_pestat] mean and std.dev: (121.89, 44.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 364)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4000832 reads in 316.039 CPU sec, 16.177 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (12, 1597207, 7, 12)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 175, 253)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 539)\n",
      "[M::mem_pestat] mean and std.dev: (184.83, 97.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 682)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (112, 151, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 379)\n",
      "[M::mem_pestat] mean and std.dev: (159.73, 64.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 468)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 167, 214)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 422)\n",
      "[M::mem_pestat] mean and std.dev: (163.36, 87.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 526)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000810 sequences (240000050 bp)...\n",
      "[M::mem_process_seqs] Processed 4000818 reads in 316.278 CPU sec, 14.116 real sec\n",
      "[M::process] read 4000804 sequences (240000052 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 1598984, 15, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (88, 111, 144)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 256)\n",
      "[M::mem_pestat] mean and std.dev: (99.61, 33.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 312)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 201)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 381)\n",
      "[M::mem_pestat] mean and std.dev: (159.43, 64.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 471)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 254, 1655)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4691)\n",
      "[M::mem_pestat] mean and std.dev: (612.62, 1087.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6209)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 4000810 reads in 316.004 CPU sec, 17.472 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (20, 1599058, 20, 8)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 116, 171)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 369)\n",
      "[M::mem_pestat] mean and std.dev: (105.29, 54.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 468)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 150, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.82, 64.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (172, 228, 551)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1309)\n",
      "[M::mem_pestat] mean and std.dev: (260.65, 183.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1688)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::process] read 4000824 sequences (240000002 bp)...\n",
      "[M::mem_process_seqs] Processed 4000804 reads in 318.262 CPU sec, 13.891 real sec\n",
      "[M::process] read 4000824 sequences (240000008 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (21, 1598683, 10, 14)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (62, 120, 179)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 413)\n",
      "[M::mem_pestat] mean and std.dev: (123.10, 64.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 530)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 200)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 378)\n",
      "[M::mem_pestat] mean and std.dev: (158.51, 64.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 467)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (36, 196, 249)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 675)\n",
      "[M::mem_pestat] mean and std.dev: (116.25, 88.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 888)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (78, 164, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 438)\n",
      "[M::mem_pestat] mean and std.dev: (155.00, 72.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 558)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 4000824 reads in 315.193 CPU sec, 16.494 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (39, 1598715, 16, 11)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (101, 137, 188)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 362)\n",
      "[M::mem_pestat] mean and std.dev: (137.69, 63.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 449)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (111, 149, 199)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 375)\n",
      "[M::mem_pestat] mean and std.dev: (158.03, 63.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 463)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (398, 657, 1373)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3323)\n",
      "[M::mem_pestat] mean and std.dev: (827.67, 755.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4298)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (92, 115, 156)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 284)\n",
      "[M::mem_pestat] mean and std.dev: (114.30, 32.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 348)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::process] read 4000822 sequences (240000102 bp)...\n",
      "[M::mem_process_seqs] Processed 4000824 reads in 306.455 CPU sec, 13.574 real sec\n",
      "[M::process] read 4000844 sequences (240000030 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (39, 1599456, 10, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (72, 102, 171)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 369)\n",
      "[M::mem_pestat] mean and std.dev: (129.74, 82.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 468)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 149, 198)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 374)\n",
      "[M::mem_pestat] mean and std.dev: (157.51, 63.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 462)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 566, 765)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2051)\n",
      "[M::mem_pestat] mean and std.dev: (341.00, 278.89)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2694)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 169, 206)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 368)\n",
      "[M::mem_pestat] mean and std.dev: (166.77, 52.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 449)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 4000822 reads in 317.839 CPU sec, 17.231 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (50, 1598542, 17, 7)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (78, 102, 168)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 348)\n",
      "[M::mem_pestat] mean and std.dev: (120.69, 61.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 438)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 148, 197)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 371)\n",
      "[M::mem_pestat] mean and std.dev: (156.89, 63.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 458)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (154, 230, 593)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1471)\n",
      "[M::mem_pestat] mean and std.dev: (298.20, 227.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1910)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::process] read 2731564 sequences (163861026 bp)...\n",
      "[M::mem_process_seqs] Processed 4000844 reads in 304.650 CPU sec, 13.699 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (28, 1091871, 12, 13)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (81, 126, 186)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 396)\n",
      "[M::mem_pestat] mean and std.dev: (133.33, 66.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 501)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (110, 148, 197)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 371)\n",
      "[M::mem_pestat] mean and std.dev: (156.74, 62.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 458)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (187, 306, 1043)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2755)\n",
      "[M::mem_pestat] mean and std.dev: (620.42, 732.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 3611)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (108, 128, 175)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 309)\n",
      "[M::mem_pestat] mean and std.dev: (143.85, 56.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 376)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 2731564 reads in 213.495 CPU sec, 12.308 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 24 /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.fa /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz\n",
      "[main] Real time: 505.043 sec; CPU: 9936.918 sec\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "!bwa mem -t 24 ~/scratch/Project2/ReadAlignment/hg38.fa ~/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz ~/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz> ~/scratch/Project2/ReadAlignment/SRR099988.sam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd9618b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "for index, I used bwtsw for algorithm as it works well with whole human genome data\n",
    "bwa: burrows-Wheeler Aligner software packages to map non-reference genome\n",
    "mem: maximal exact match\n",
    "-t 24: -t argument calls the number of CPU cores for the alinment process. In this command, we use 24 CPU cores for the alignment.\n",
    "\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cf2aa1de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mm_idx_gen::33.720*1.52] collected minimizers\n",
      "[M::mm_idx_gen::43.669*1.85] sorted minimizers\n",
      "[M::main::49.499*1.75] loaded/built the index for 455 target sequence(s)\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 455\n",
      "[M::mm_idx_stat::50.147*1.74] distinct minimizers: 100202295 (37.96% are singletons); average occurrences: 5.732; average spacing: 5.587; total length: 3209286105\n",
      "[M::main] Version: 2.26-r1175\n",
      "[M::main] CMD: minimap2 -d /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.mmi /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.fa\n",
      "[M::main] Real time: 50.220 sec; CPU: 87.282 sec; Peak RSS: 11.899 GB\n"
     ]
    }
   ],
   "source": [
    "!minimap2 -d ~/scratch/Project2/ReadAlignment/hg38.mmi ~/scratch/Project2/ReadAlignment/hg38.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b178269e",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "-d: this option will save the index of target fastsa file as a new file (in our case, that'll be an .mmi file)\n",
    "the first path is the path to an output file, the next path is the one to an input file\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "903dba01",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::main::4.543*1.00] loaded/built the index for 455 target sequence(s)\n",
      "[M::mm_mapopt_update::5.560*1.00] mid_occ = 728\n",
      "[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 455\n",
      "[M::mm_idx_stat::6.195*1.00] distinct minimizers: 100202295 (37.96% are singletons); average occurrences: 5.732; average spacing: 5.587; total length: 3209286105\n",
      "[M::worker_pipeline::79.839*2.86] mapped 8335025 sequences\n",
      "[M::worker_pipeline::132.906*3.05] mapped 8335054 sequences\n",
      "[M::worker_pipeline::185.033*3.13] mapped 8335037 sequences\n",
      "[M::worker_pipeline::237.960*3.18] mapped 8335044 sequences\n",
      "[M::worker_pipeline::290.295*3.21] mapped 8335009 sequences\n",
      "[M::worker_pipeline::342.826*3.23] mapped 8335038 sequences\n",
      "[M::worker_pipeline::393.841*3.24] mapped 8335028 sequences\n",
      "[M::worker_pipeline::421.268*3.19] mapped 5033229 sequences\n",
      "[M::worker_pipeline::491.561*3.17] mapped 8335025 sequences\n",
      "[M::worker_pipeline::543.664*3.18] mapped 8335054 sequences\n",
      "[M::worker_pipeline::595.517*3.20] mapped 8335037 sequences\n",
      "[M::worker_pipeline::647.357*3.21] mapped 8335044 sequences\n",
      "[M::worker_pipeline::700.243*3.22] mapped 8335009 sequences\n",
      "[M::worker_pipeline::749.962*3.23] mapped 8335038 sequences\n",
      "[M::worker_pipeline::801.767*3.23] mapped 8335028 sequences\n",
      "[M::worker_pipeline::829.369*3.21] mapped 5033229 sequences\n",
      "[M::main] Version: 2.26-r1175\n",
      "[M::main] CMD: minimap2 -c -a /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.mmi /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz\n",
      "[M::main] Real time: 829.538 sec; CPU: 2661.410 sec; Peak RSS: 13.993 GB\n"
     ]
    }
   ],
   "source": [
    "!minimap2 -c -a ~/scratch/Project2/ReadAlignment/hg38.mmi ~/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz ~/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz> ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bef7051f",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "-c: an input/output option that generates Compact Idiosyncratic Gapped Alignment Report (CIGAR).\n",
    "-a: an input/output option that creates CIGAR and output alignments in the SAM format.\n",
    "the first path is minimap indexed hg38 reference genome\n",
    "the second and first paths are two outputs for the alignment\n",
    "the path after \">\" refers to an output\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1a13ead4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "176131465 /home/hice1/ynitta6/scratch/Project2/ReadAlignment/SRR099988_minimap.sam\r\n"
     ]
    }
   ],
   "source": [
    "!wc -l ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam\n",
    "\n",
    "'this file has 176131465 total lines'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ce58e70b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "176131008\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view -c ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam\n",
    "'this file has 176131008 lines for alignment records'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f530460",
   "metadata": {},
   "source": [
    "# **3. Post Read alignment**\n",
    "\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4b469ea4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Program: samtools (Tools for alignments in the SAM format)\r\n",
      "Version: 1.17 (using htslib 1.18)\r\n",
      "\r\n",
      "Usage:   samtools <command> [options]\r\n",
      "\r\n",
      "Commands:\r\n",
      "  -- Indexing\r\n",
      "     dict           create a sequence dictionary file\r\n",
      "     faidx          index/extract FASTA\r\n",
      "     fqidx          index/extract FASTQ\r\n",
      "     index          index alignment\r\n",
      "\r\n",
      "  -- Editing\r\n",
      "     calmd          recalculate MD/NM tags and '=' bases\r\n",
      "     fixmate        fix mate information\r\n",
      "     reheader       replace BAM header\r\n",
      "     targetcut      cut fosmid regions (for fosmid pool only)\r\n",
      "     addreplacerg   adds or replaces RG tags\r\n",
      "     markdup        mark duplicates\r\n",
      "     ampliconclip   clip oligos from the end of reads\r\n",
      "\r\n",
      "  -- File operations\r\n",
      "     collate        shuffle and group alignments by name\r\n",
      "     cat            concatenate BAMs\r\n",
      "     consensus      produce a consensus Pileup/FASTA/FASTQ\r\n",
      "     merge          merge sorted alignments\r\n",
      "     mpileup        multi-way pileup\r\n",
      "     sort           sort alignment file\r\n",
      "     split          splits a file by read group\r\n",
      "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\r\n",
      "     fastq          converts a BAM to a FASTQ\r\n",
      "     fasta          converts a BAM to a FASTA\r\n",
      "     import         Converts FASTA or FASTQ files to SAM/BAM/CRAM\r\n",
      "     reference      Generates a reference from aligned data\r\n",
      "     reset          Reverts aligner changes in reads\r\n",
      "\r\n",
      "  -- Statistics\r\n",
      "     bedcov         read depth per BED region\r\n",
      "     coverage       alignment depth and percent coverage\r\n",
      "     depth          compute the depth\r\n",
      "     flagstat       simple stats\r\n",
      "     idxstats       BAM index stats\r\n",
      "     cram-size      list CRAM Content-ID and Data-Series sizes\r\n",
      "     phase          phase heterozygotes\r\n",
      "     stats          generate stats (former bamcheck)\r\n",
      "     ampliconstats  generate amplicon specific stats\r\n",
      "\r\n",
      "  -- Viewing\r\n",
      "     flags          explain BAM flags\r\n",
      "     head           header viewer\r\n",
      "     tview          text alignment viewer\r\n",
      "     view           SAM<->BAM<->CRAM conversion\r\n",
      "     depad          convert padded BAM to unpadded BAM\r\n",
      "     samples        list the samples in a set of SAM/BAM/CRAM files\r\n",
      "\r\n",
      "  -- Misc\r\n",
      "     help [cmd]     display this help message or help for [cmd]\r\n",
      "     version        detailed version information\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a700b0dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@HD\tVN:1.6\tSO:unsorted\tGO:query\r\n",
      "@SQ\tSN:chr1\tLN:248956422\r\n",
      "@SQ\tSN:chr10\tLN:133797422\r\n",
      "@SQ\tSN:chr11\tLN:135086622\r\n",
      "@SQ\tSN:chr11_KI270721v1_random\tLN:100316\r\n",
      "@SQ\tSN:chr12\tLN:133275309\r\n",
      "@SQ\tSN:chr13\tLN:114364328\r\n",
      "@SQ\tSN:chr14\tLN:107043718\r\n",
      "@SQ\tSN:chr14_GL000009v2_random\tLN:201709\r\n",
      "@SQ\tSN:chr14_GL000225v1_random\tLN:211173\r\n",
      "@SQ\tSN:chr14_KI270722v1_random\tLN:194050\r\n",
      "@SQ\tSN:chr14_GL000194v1_random\tLN:191469\r\n",
      "@SQ\tSN:chr14_KI270723v1_random\tLN:38115\r\n",
      "@SQ\tSN:chr14_KI270724v1_random\tLN:39555\r\n",
      "@SQ\tSN:chr14_KI270725v1_random\tLN:172810\r\n",
      "@SQ\tSN:chr14_KI270726v1_random\tLN:43739\r\n",
      "@SQ\tSN:chr15\tLN:101991189\r\n",
      "@SQ\tSN:chr15_KI270727v1_random\tLN:448248\r\n",
      "@SQ\tSN:chr16\tLN:90338345\r\n",
      "@SQ\tSN:chr16_KI270728v1_random\tLN:1872759\r\n",
      "@SQ\tSN:chr17\tLN:83257441\r\n",
      "@SQ\tSN:chr17_GL000205v2_random\tLN:185591\r\n",
      "@SQ\tSN:chr17_KI270729v1_random\tLN:280839\r\n",
      "@SQ\tSN:chr17_KI270730v1_random\tLN:112551\r\n",
      "@SQ\tSN:chr18\tLN:80373285\r\n",
      "@SQ\tSN:chr19\tLN:58617616\r\n",
      "@SQ\tSN:chr1_KI270706v1_random\tLN:175055\r\n",
      "@SQ\tSN:chr1_KI270707v1_random\tLN:32032\r\n",
      "@SQ\tSN:chr1_KI270708v1_random\tLN:127682\r\n",
      "@SQ\tSN:chr1_KI270709v1_random\tLN:66860\r\n",
      "@SQ\tSN:chr1_KI270710v1_random\tLN:40176\r\n",
      "@SQ\tSN:chr1_KI270711v1_random\tLN:42210\r\n",
      "@SQ\tSN:chr1_KI270712v1_random\tLN:176043\r\n",
      "@SQ\tSN:chr1_KI270713v1_random\tLN:40745\r\n",
      "@SQ\tSN:chr1_KI270714v1_random\tLN:41717\r\n",
      "@SQ\tSN:chr2\tLN:242193529\r\n",
      "@SQ\tSN:chr20\tLN:64444167\r\n",
      "@SQ\tSN:chr21\tLN:46709983\r\n",
      "@SQ\tSN:chr22\tLN:50818468\r\n",
      "@SQ\tSN:chr22_KI270731v1_random\tLN:150754\r\n",
      "@SQ\tSN:chr22_KI270732v1_random\tLN:41543\r\n",
      "@SQ\tSN:chr22_KI270733v1_random\tLN:179772\r\n",
      "@SQ\tSN:chr22_KI270734v1_random\tLN:165050\r\n",
      "@SQ\tSN:chr22_KI270735v1_random\tLN:42811\r\n",
      "@SQ\tSN:chr22_KI270736v1_random\tLN:181920\r\n",
      "@SQ\tSN:chr22_KI270737v1_random\tLN:103838\r\n",
      "@SQ\tSN:chr22_KI270738v1_random\tLN:99375\r\n",
      "@SQ\tSN:chr22_KI270739v1_random\tLN:73985\r\n",
      "@SQ\tSN:chr2_KI270715v1_random\tLN:161471\r\n",
      "@SQ\tSN:chr2_KI270716v1_random\tLN:153799\r\n",
      "@SQ\tSN:chr3\tLN:198295559\r\n",
      "@SQ\tSN:chr3_GL000221v1_random\tLN:155397\r\n",
      "@SQ\tSN:chr4\tLN:190214555\r\n",
      "@SQ\tSN:chr4_GL000008v2_random\tLN:209709\r\n",
      "@SQ\tSN:chr5\tLN:181538259\r\n",
      "@SQ\tSN:chr5_GL000208v1_random\tLN:92689\r\n",
      "@SQ\tSN:chr6\tLN:170805979\r\n",
      "@SQ\tSN:chr7\tLN:159345973\r\n",
      "@SQ\tSN:chr8\tLN:145138636\r\n",
      "@SQ\tSN:chr9\tLN:138394717\r\n",
      "@SQ\tSN:chr9_KI270717v1_random\tLN:40062\r\n",
      "@SQ\tSN:chr9_KI270718v1_random\tLN:38054\r\n",
      "@SQ\tSN:chr9_KI270719v1_random\tLN:176845\r\n",
      "@SQ\tSN:chr9_KI270720v1_random\tLN:39050\r\n",
      "@SQ\tSN:chr1_KI270762v1_alt\tLN:354444\r\n",
      "@SQ\tSN:chr1_KI270766v1_alt\tLN:256271\r\n",
      "@SQ\tSN:chr1_KI270760v1_alt\tLN:109528\r\n",
      "@SQ\tSN:chr1_KI270765v1_alt\tLN:185285\r\n",
      "@SQ\tSN:chr1_GL383518v1_alt\tLN:182439\r\n",
      "@SQ\tSN:chr1_GL383519v1_alt\tLN:110268\r\n",
      "@SQ\tSN:chr1_GL383520v2_alt\tLN:366580\r\n",
      "@SQ\tSN:chr1_KI270764v1_alt\tLN:50258\r\n",
      "@SQ\tSN:chr1_KI270763v1_alt\tLN:911658\r\n",
      "@SQ\tSN:chr1_KI270759v1_alt\tLN:425601\r\n",
      "@SQ\tSN:chr1_KI270761v1_alt\tLN:165834\r\n",
      "@SQ\tSN:chr2_KI270770v1_alt\tLN:136240\r\n",
      "@SQ\tSN:chr2_KI270773v1_alt\tLN:70887\r\n",
      "@SQ\tSN:chr2_KI270774v1_alt\tLN:223625\r\n",
      "@SQ\tSN:chr2_KI270769v1_alt\tLN:120616\r\n",
      "@SQ\tSN:chr2_GL383521v1_alt\tLN:143390\r\n",
      "@SQ\tSN:chr2_KI270772v1_alt\tLN:133041\r\n",
      "@SQ\tSN:chr2_KI270775v1_alt\tLN:138019\r\n",
      "@SQ\tSN:chr2_KI270771v1_alt\tLN:110395\r\n",
      "@SQ\tSN:chr2_KI270768v1_alt\tLN:110099\r\n",
      "@SQ\tSN:chr2_GL582966v2_alt\tLN:96131\r\n",
      "@SQ\tSN:chr2_GL383522v1_alt\tLN:123821\r\n",
      "@SQ\tSN:chr2_KI270776v1_alt\tLN:174166\r\n",
      "@SQ\tSN:chr2_KI270767v1_alt\tLN:161578\r\n",
      "@SQ\tSN:chr3_JH636055v2_alt\tLN:173151\r\n",
      "@SQ\tSN:chr3_KI270783v1_alt\tLN:109187\r\n",
      "@SQ\tSN:chr3_KI270780v1_alt\tLN:224108\r\n",
      "@SQ\tSN:chr3_GL383526v1_alt\tLN:180671\r\n",
      "@SQ\tSN:chr3_KI270777v1_alt\tLN:173649\r\n",
      "@SQ\tSN:chr3_KI270778v1_alt\tLN:248252\r\n",
      "@SQ\tSN:chr3_KI270781v1_alt\tLN:113034\r\n",
      "@SQ\tSN:chr3_KI270779v1_alt\tLN:205312\r\n",
      "@SQ\tSN:chr3_KI270782v1_alt\tLN:162429\r\n",
      "@SQ\tSN:chr3_KI270784v1_alt\tLN:184404\r\n",
      "@SQ\tSN:chr4_KI270790v1_alt\tLN:220246\r\n",
      "@SQ\tSN:chr4_GL383528v1_alt\tLN:376187\r\n",
      "@SQ\tSN:chr4_KI270787v1_alt\tLN:111943\r\n",
      "@SQ\tSN:chr4_GL000257v2_alt\tLN:586476\r\n",
      "@SQ\tSN:chr4_KI270788v1_alt\tLN:158965\r\n",
      "@SQ\tSN:chr4_GL383527v1_alt\tLN:164536\r\n",
      "@SQ\tSN:chr4_KI270785v1_alt\tLN:119912\r\n",
      "@SQ\tSN:chr4_KI270789v1_alt\tLN:205944\r\n",
      "@SQ\tSN:chr4_KI270786v1_alt\tLN:244096\r\n",
      "@SQ\tSN:chr5_KI270793v1_alt\tLN:126136\r\n",
      "@SQ\tSN:chr5_KI270792v1_alt\tLN:179043\r\n",
      "@SQ\tSN:chr5_KI270791v1_alt\tLN:195710\r\n",
      "@SQ\tSN:chr5_GL383532v1_alt\tLN:82728\r\n",
      "@SQ\tSN:chr5_GL949742v1_alt\tLN:226852\r\n",
      "@SQ\tSN:chr5_KI270794v1_alt\tLN:164558\r\n",
      "@SQ\tSN:chr5_GL339449v2_alt\tLN:1612928\r\n",
      "@SQ\tSN:chr5_GL383530v1_alt\tLN:101241\r\n",
      "@SQ\tSN:chr5_KI270796v1_alt\tLN:172708\r\n",
      "@SQ\tSN:chr5_GL383531v1_alt\tLN:173459\r\n",
      "@SQ\tSN:chr5_KI270795v1_alt\tLN:131892\r\n",
      "@SQ\tSN:chr6_GL000250v2_alt\tLN:4672374\r\n",
      "@SQ\tSN:chr6_KI270800v1_alt\tLN:175808\r\n",
      "@SQ\tSN:chr6_KI270799v1_alt\tLN:152148\r\n",
      "@SQ\tSN:chr6_GL383533v1_alt\tLN:124736\r\n",
      "@SQ\tSN:chr6_KI270801v1_alt\tLN:870480\r\n",
      "@SQ\tSN:chr6_KI270802v1_alt\tLN:75005\r\n",
      "@SQ\tSN:chr6_KB021644v2_alt\tLN:185823\r\n",
      "@SQ\tSN:chr6_KI270797v1_alt\tLN:197536\r\n",
      "@SQ\tSN:chr6_KI270798v1_alt\tLN:271782\r\n",
      "@SQ\tSN:chr7_KI270804v1_alt\tLN:157952\r\n",
      "@SQ\tSN:chr7_KI270809v1_alt\tLN:209586\r\n",
      "@SQ\tSN:chr7_KI270806v1_alt\tLN:158166\r\n",
      "@SQ\tSN:chr7_GL383534v2_alt\tLN:119183\r\n",
      "@SQ\tSN:chr7_KI270803v1_alt\tLN:1111570\r\n",
      "@SQ\tSN:chr7_KI270808v1_alt\tLN:271455\r\n",
      "@SQ\tSN:chr7_KI270807v1_alt\tLN:126434\r\n",
      "@SQ\tSN:chr7_KI270805v1_alt\tLN:209988\r\n",
      "@SQ\tSN:chr8_KI270818v1_alt\tLN:145606\r\n",
      "@SQ\tSN:chr8_KI270812v1_alt\tLN:282736\r\n",
      "@SQ\tSN:chr8_KI270811v1_alt\tLN:292436\r\n",
      "@SQ\tSN:chr8_KI270821v1_alt\tLN:985506\r\n",
      "@SQ\tSN:chr8_KI270813v1_alt\tLN:300230\r\n",
      "@SQ\tSN:chr8_KI270822v1_alt\tLN:624492\r\n",
      "@SQ\tSN:chr8_KI270814v1_alt\tLN:141812\r\n",
      "@SQ\tSN:chr8_KI270810v1_alt\tLN:374415\r\n",
      "@SQ\tSN:chr8_KI270819v1_alt\tLN:133535\r\n",
      "@SQ\tSN:chr8_KI270820v1_alt\tLN:36640\r\n",
      "@SQ\tSN:chr8_KI270817v1_alt\tLN:158983\r\n",
      "@SQ\tSN:chr8_KI270816v1_alt\tLN:305841\r\n",
      "@SQ\tSN:chr8_KI270815v1_alt\tLN:132244\r\n",
      "@SQ\tSN:chr9_GL383539v1_alt\tLN:162988\r\n",
      "@SQ\tSN:chr9_GL383540v1_alt\tLN:71551\r\n",
      "@SQ\tSN:chr9_GL383541v1_alt\tLN:171286\r\n",
      "@SQ\tSN:chr9_GL383542v1_alt\tLN:60032\r\n",
      "@SQ\tSN:chr9_KI270823v1_alt\tLN:439082\r\n",
      "@SQ\tSN:chr10_GL383545v1_alt\tLN:179254\r\n",
      "@SQ\tSN:chr10_KI270824v1_alt\tLN:181496\r\n",
      "@SQ\tSN:chr10_GL383546v1_alt\tLN:309802\r\n",
      "@SQ\tSN:chr10_KI270825v1_alt\tLN:188315\r\n",
      "@SQ\tSN:chr11_KI270832v1_alt\tLN:210133\r\n",
      "@SQ\tSN:chr11_KI270830v1_alt\tLN:177092\r\n",
      "@SQ\tSN:chr11_KI270831v1_alt\tLN:296895\r\n",
      "@SQ\tSN:chr11_KI270829v1_alt\tLN:204059\r\n",
      "@SQ\tSN:chr11_GL383547v1_alt\tLN:154407\r\n",
      "@SQ\tSN:chr11_JH159136v1_alt\tLN:200998\r\n",
      "@SQ\tSN:chr11_JH159137v1_alt\tLN:191409\r\n",
      "@SQ\tSN:chr11_KI270827v1_alt\tLN:67707\r\n",
      "@SQ\tSN:chr11_KI270826v1_alt\tLN:186169\r\n",
      "@SQ\tSN:chr12_GL877875v1_alt\tLN:167313\r\n",
      "@SQ\tSN:chr12_GL877876v1_alt\tLN:408271\r\n",
      "@SQ\tSN:chr12_KI270837v1_alt\tLN:40090\r\n",
      "@SQ\tSN:chr12_GL383549v1_alt\tLN:120804\r\n",
      "@SQ\tSN:chr12_KI270835v1_alt\tLN:238139\r\n",
      "@SQ\tSN:chr12_GL383550v2_alt\tLN:169178\r\n",
      "@SQ\tSN:chr12_GL383552v1_alt\tLN:138655\r\n",
      "@SQ\tSN:chr12_GL383553v2_alt\tLN:152874\r\n",
      "@SQ\tSN:chr12_KI270834v1_alt\tLN:119498\r\n",
      "@SQ\tSN:chr12_GL383551v1_alt\tLN:184319\r\n",
      "@SQ\tSN:chr12_KI270833v1_alt\tLN:76061\r\n",
      "@SQ\tSN:chr12_KI270836v1_alt\tLN:56134\r\n",
      "@SQ\tSN:chr13_KI270840v1_alt\tLN:191684\r\n",
      "@SQ\tSN:chr13_KI270839v1_alt\tLN:180306\r\n",
      "@SQ\tSN:chr13_KI270843v1_alt\tLN:103832\r\n",
      "@SQ\tSN:chr13_KI270841v1_alt\tLN:169134\r\n",
      "@SQ\tSN:chr13_KI270838v1_alt\tLN:306913\r\n",
      "@SQ\tSN:chr13_KI270842v1_alt\tLN:37287\r\n",
      "@SQ\tSN:chr14_KI270844v1_alt\tLN:322166\r\n",
      "@SQ\tSN:chr14_KI270847v1_alt\tLN:1511111\r\n",
      "@SQ\tSN:chr14_KI270845v1_alt\tLN:180703\r\n",
      "@SQ\tSN:chr14_KI270846v1_alt\tLN:1351393\r\n",
      "@SQ\tSN:chr15_KI270852v1_alt\tLN:478999\r\n",
      "@SQ\tSN:chr15_KI270851v1_alt\tLN:263054\r\n",
      "@SQ\tSN:chr15_KI270848v1_alt\tLN:327382\r\n",
      "@SQ\tSN:chr15_GL383554v1_alt\tLN:296527\r\n",
      "@SQ\tSN:chr15_KI270849v1_alt\tLN:244917\r\n",
      "@SQ\tSN:chr15_GL383555v2_alt\tLN:388773\r\n",
      "@SQ\tSN:chr15_KI270850v1_alt\tLN:430880\r\n",
      "@SQ\tSN:chr16_KI270854v1_alt\tLN:134193\r\n",
      "@SQ\tSN:chr16_KI270856v1_alt\tLN:63982\r\n",
      "@SQ\tSN:chr16_KI270855v1_alt\tLN:232857\r\n",
      "@SQ\tSN:chr16_KI270853v1_alt\tLN:2659700\r\n",
      "@SQ\tSN:chr16_GL383556v1_alt\tLN:192462\r\n",
      "@SQ\tSN:chr16_GL383557v1_alt\tLN:89672\r\n",
      "@SQ\tSN:chr17_GL383563v3_alt\tLN:375691\r\n",
      "@SQ\tSN:chr17_KI270862v1_alt\tLN:391357\r\n",
      "@SQ\tSN:chr17_KI270861v1_alt\tLN:196688\r\n",
      "@SQ\tSN:chr17_KI270857v1_alt\tLN:2877074\r\n",
      "@SQ\tSN:chr17_JH159146v1_alt\tLN:278131\r\n",
      "@SQ\tSN:chr17_JH159147v1_alt\tLN:70345\r\n",
      "@SQ\tSN:chr17_GL383564v2_alt\tLN:133151\r\n",
      "@SQ\tSN:chr17_GL000258v2_alt\tLN:1821992\r\n",
      "@SQ\tSN:chr17_GL383565v1_alt\tLN:223995\r\n",
      "@SQ\tSN:chr17_KI270858v1_alt\tLN:235827\r\n",
      "@SQ\tSN:chr17_KI270859v1_alt\tLN:108763\r\n",
      "@SQ\tSN:chr17_GL383566v1_alt\tLN:90219\r\n",
      "@SQ\tSN:chr17_KI270860v1_alt\tLN:178921\r\n",
      "@SQ\tSN:chr18_KI270864v1_alt\tLN:111737\r\n",
      "@SQ\tSN:chr18_GL383567v1_alt\tLN:289831\r\n",
      "@SQ\tSN:chr18_GL383570v1_alt\tLN:164789\r\n",
      "@SQ\tSN:chr18_GL383571v1_alt\tLN:198278\r\n",
      "@SQ\tSN:chr18_GL383568v1_alt\tLN:104552\r\n",
      "@SQ\tSN:chr18_GL383569v1_alt\tLN:167950\r\n",
      "@SQ\tSN:chr18_GL383572v1_alt\tLN:159547\r\n",
      "@SQ\tSN:chr18_KI270863v1_alt\tLN:167999\r\n",
      "@SQ\tSN:chr19_KI270868v1_alt\tLN:61734\r\n",
      "@SQ\tSN:chr19_KI270865v1_alt\tLN:52969\r\n",
      "@SQ\tSN:chr19_GL383573v1_alt\tLN:385657\r\n",
      "@SQ\tSN:chr19_GL383575v2_alt\tLN:170222\r\n",
      "@SQ\tSN:chr19_GL383576v1_alt\tLN:188024\r\n",
      "@SQ\tSN:chr19_GL383574v1_alt\tLN:155864\r\n",
      "@SQ\tSN:chr19_KI270866v1_alt\tLN:43156\r\n",
      "@SQ\tSN:chr19_KI270867v1_alt\tLN:233762\r\n",
      "@SQ\tSN:chr19_GL949746v1_alt\tLN:987716\r\n",
      "@SQ\tSN:chr20_GL383577v2_alt\tLN:128386\r\n",
      "@SQ\tSN:chr20_KI270869v1_alt\tLN:118774\r\n",
      "@SQ\tSN:chr20_KI270871v1_alt\tLN:58661\r\n",
      "@SQ\tSN:chr20_KI270870v1_alt\tLN:183433\r\n",
      "@SQ\tSN:chr21_GL383578v2_alt\tLN:63917\r\n",
      "@SQ\tSN:chr21_KI270874v1_alt\tLN:166743\r\n",
      "@SQ\tSN:chr21_KI270873v1_alt\tLN:143900\r\n",
      "@SQ\tSN:chr21_GL383579v2_alt\tLN:201197\r\n",
      "@SQ\tSN:chr21_GL383580v2_alt\tLN:74653\r\n",
      "@SQ\tSN:chr21_GL383581v2_alt\tLN:116689\r\n",
      "@SQ\tSN:chr21_KI270872v1_alt\tLN:82692\r\n",
      "@SQ\tSN:chr22_KI270875v1_alt\tLN:259914\r\n",
      "@SQ\tSN:chr22_KI270878v1_alt\tLN:186262\r\n",
      "@SQ\tSN:chr22_KI270879v1_alt\tLN:304135\r\n",
      "@SQ\tSN:chr22_KI270876v1_alt\tLN:263666\r\n",
      "@SQ\tSN:chr22_KI270877v1_alt\tLN:101331\r\n",
      "@SQ\tSN:chr22_GL383583v2_alt\tLN:96924\r\n",
      "@SQ\tSN:chr22_GL383582v2_alt\tLN:162811\r\n",
      "@SQ\tSN:chrX_KI270880v1_alt\tLN:284869\r\n",
      "@SQ\tSN:chrX_KI270881v1_alt\tLN:144206\r\n",
      "@SQ\tSN:chr19_KI270882v1_alt\tLN:248807\r\n",
      "@SQ\tSN:chr19_KI270883v1_alt\tLN:170399\r\n",
      "@SQ\tSN:chr19_KI270884v1_alt\tLN:157053\r\n",
      "@SQ\tSN:chr19_KI270885v1_alt\tLN:171027\r\n",
      "@SQ\tSN:chr19_KI270886v1_alt\tLN:204239\r\n",
      "@SQ\tSN:chr19_KI270887v1_alt\tLN:209512\r\n",
      "@SQ\tSN:chr19_KI270888v1_alt\tLN:155532\r\n",
      "@SQ\tSN:chr19_KI270889v1_alt\tLN:170698\r\n",
      "@SQ\tSN:chr19_KI270890v1_alt\tLN:184499\r\n",
      "@SQ\tSN:chr19_KI270891v1_alt\tLN:170680\r\n",
      "@SQ\tSN:chr1_KI270892v1_alt\tLN:162212\r\n",
      "@SQ\tSN:chr2_KI270894v1_alt\tLN:214158\r\n",
      "@SQ\tSN:chr2_KI270893v1_alt\tLN:161218\r\n",
      "@SQ\tSN:chr3_KI270895v1_alt\tLN:162896\r\n",
      "@SQ\tSN:chr4_KI270896v1_alt\tLN:378547\r\n",
      "@SQ\tSN:chr5_KI270897v1_alt\tLN:1144418\r\n",
      "@SQ\tSN:chr5_KI270898v1_alt\tLN:130957\r\n",
      "@SQ\tSN:chr6_GL000251v2_alt\tLN:4795265\r\n",
      "@SQ\tSN:chr7_KI270899v1_alt\tLN:190869\r\n",
      "@SQ\tSN:chr8_KI270901v1_alt\tLN:136959\r\n",
      "@SQ\tSN:chr8_KI270900v1_alt\tLN:318687\r\n",
      "@SQ\tSN:chr11_KI270902v1_alt\tLN:106711\r\n",
      "@SQ\tSN:chr11_KI270903v1_alt\tLN:214625\r\n",
      "@SQ\tSN:chr12_KI270904v1_alt\tLN:572349\r\n",
      "@SQ\tSN:chr15_KI270906v1_alt\tLN:196384\r\n",
      "@SQ\tSN:chr15_KI270905v1_alt\tLN:5161414\r\n",
      "@SQ\tSN:chr17_KI270907v1_alt\tLN:137721\r\n",
      "@SQ\tSN:chr17_KI270910v1_alt\tLN:157099\r\n",
      "@SQ\tSN:chr17_KI270909v1_alt\tLN:325800\r\n",
      "@SQ\tSN:chr17_JH159148v1_alt\tLN:88070\r\n",
      "@SQ\tSN:chr17_KI270908v1_alt\tLN:1423190\r\n",
      "@SQ\tSN:chr18_KI270912v1_alt\tLN:174061\r\n",
      "@SQ\tSN:chr18_KI270911v1_alt\tLN:157710\r\n",
      "@SQ\tSN:chr19_GL949747v2_alt\tLN:729520\r\n",
      "@SQ\tSN:chr22_KB663609v1_alt\tLN:74013\r\n",
      "@SQ\tSN:chrX_KI270913v1_alt\tLN:274009\r\n",
      "@SQ\tSN:chr19_KI270914v1_alt\tLN:205194\r\n",
      "@SQ\tSN:chr19_KI270915v1_alt\tLN:170665\r\n",
      "@SQ\tSN:chr19_KI270916v1_alt\tLN:184516\r\n",
      "@SQ\tSN:chr19_KI270917v1_alt\tLN:190932\r\n",
      "@SQ\tSN:chr19_KI270918v1_alt\tLN:123111\r\n",
      "@SQ\tSN:chr19_KI270919v1_alt\tLN:170701\r\n",
      "@SQ\tSN:chr19_KI270920v1_alt\tLN:198005\r\n",
      "@SQ\tSN:chr19_KI270921v1_alt\tLN:282224\r\n",
      "@SQ\tSN:chr19_KI270922v1_alt\tLN:187935\r\n",
      "@SQ\tSN:chr19_KI270923v1_alt\tLN:189352\r\n",
      "@SQ\tSN:chr3_KI270924v1_alt\tLN:166540\r\n",
      "@SQ\tSN:chr4_KI270925v1_alt\tLN:555799\r\n",
      "@SQ\tSN:chr6_GL000252v2_alt\tLN:4604811\r\n",
      "@SQ\tSN:chr8_KI270926v1_alt\tLN:229282\r\n",
      "@SQ\tSN:chr11_KI270927v1_alt\tLN:218612\r\n",
      "@SQ\tSN:chr19_GL949748v2_alt\tLN:1064304\r\n",
      "@SQ\tSN:chr22_KI270928v1_alt\tLN:176103\r\n",
      "@SQ\tSN:chr19_KI270929v1_alt\tLN:186203\r\n",
      "@SQ\tSN:chr19_KI270930v1_alt\tLN:200773\r\n",
      "@SQ\tSN:chr19_KI270931v1_alt\tLN:170148\r\n",
      "@SQ\tSN:chr19_KI270932v1_alt\tLN:215732\r\n",
      "@SQ\tSN:chr19_KI270933v1_alt\tLN:170537\r\n",
      "@SQ\tSN:chr19_GL000209v2_alt\tLN:177381\r\n",
      "@SQ\tSN:chr3_KI270934v1_alt\tLN:163458\r\n",
      "@SQ\tSN:chr6_GL000253v2_alt\tLN:4677643\r\n",
      "@SQ\tSN:chr19_GL949749v2_alt\tLN:1091841\r\n",
      "@SQ\tSN:chr3_KI270935v1_alt\tLN:197351\r\n",
      "@SQ\tSN:chr6_GL000254v2_alt\tLN:4827813\r\n",
      "@SQ\tSN:chr19_GL949750v2_alt\tLN:1066390\r\n",
      "@SQ\tSN:chr3_KI270936v1_alt\tLN:164170\r\n",
      "@SQ\tSN:chr6_GL000255v2_alt\tLN:4606388\r\n",
      "@SQ\tSN:chr19_GL949751v2_alt\tLN:1002683\r\n",
      "@SQ\tSN:chr3_KI270937v1_alt\tLN:165607\r\n",
      "@SQ\tSN:chr6_GL000256v2_alt\tLN:4929269\r\n",
      "@SQ\tSN:chr19_GL949752v1_alt\tLN:987100\r\n",
      "@SQ\tSN:chr6_KI270758v1_alt\tLN:76752\r\n",
      "@SQ\tSN:chr19_GL949753v2_alt\tLN:796479\r\n",
      "@SQ\tSN:chr19_KI270938v1_alt\tLN:1066800\r\n",
      "@SQ\tSN:chrM\tLN:16569\r\n",
      "@SQ\tSN:chrUn_KI270302v1\tLN:2274\r\n",
      "@SQ\tSN:chrUn_KI270304v1\tLN:2165\r\n",
      "@SQ\tSN:chrUn_KI270303v1\tLN:1942\r\n",
      "@SQ\tSN:chrUn_KI270305v1\tLN:1472\r\n",
      "@SQ\tSN:chrUn_KI270322v1\tLN:21476\r\n",
      "@SQ\tSN:chrUn_KI270320v1\tLN:4416\r\n",
      "@SQ\tSN:chrUn_KI270310v1\tLN:1201\r\n",
      "@SQ\tSN:chrUn_KI270316v1\tLN:1444\r\n",
      "@SQ\tSN:chrUn_KI270315v1\tLN:2276\r\n",
      "@SQ\tSN:chrUn_KI270312v1\tLN:998\r\n",
      "@SQ\tSN:chrUn_KI270311v1\tLN:12399\r\n",
      "@SQ\tSN:chrUn_KI270317v1\tLN:37690\r\n",
      "@SQ\tSN:chrUn_KI270412v1\tLN:1179\r\n",
      "@SQ\tSN:chrUn_KI270411v1\tLN:2646\r\n",
      "@SQ\tSN:chrUn_KI270414v1\tLN:2489\r\n",
      "@SQ\tSN:chrUn_KI270419v1\tLN:1029\r\n",
      "@SQ\tSN:chrUn_KI270418v1\tLN:2145\r\n",
      "@SQ\tSN:chrUn_KI270420v1\tLN:2321\r\n",
      "@SQ\tSN:chrUn_KI270424v1\tLN:2140\r\n",
      "@SQ\tSN:chrUn_KI270417v1\tLN:2043\r\n",
      "@SQ\tSN:chrUn_KI270422v1\tLN:1445\r\n",
      "@SQ\tSN:chrUn_KI270423v1\tLN:981\r\n",
      "@SQ\tSN:chrUn_KI270425v1\tLN:1884\r\n",
      "@SQ\tSN:chrUn_KI270429v1\tLN:1361\r\n",
      "@SQ\tSN:chrUn_KI270442v1\tLN:392061\r\n",
      "@SQ\tSN:chrUn_KI270466v1\tLN:1233\r\n",
      "@SQ\tSN:chrUn_KI270465v1\tLN:1774\r\n",
      "@SQ\tSN:chrUn_KI270467v1\tLN:3920\r\n",
      "@SQ\tSN:chrUn_KI270435v1\tLN:92983\r\n",
      "@SQ\tSN:chrUn_KI270438v1\tLN:112505\r\n",
      "@SQ\tSN:chrUn_KI270468v1\tLN:4055\r\n",
      "@SQ\tSN:chrUn_KI270510v1\tLN:2415\r\n",
      "@SQ\tSN:chrUn_KI270509v1\tLN:2318\r\n",
      "@SQ\tSN:chrUn_KI270518v1\tLN:2186\r\n",
      "@SQ\tSN:chrUn_KI270508v1\tLN:1951\r\n",
      "@SQ\tSN:chrUn_KI270516v1\tLN:1300\r\n",
      "@SQ\tSN:chrUn_KI270512v1\tLN:22689\r\n",
      "@SQ\tSN:chrUn_KI270519v1\tLN:138126\r\n",
      "@SQ\tSN:chrUn_KI270522v1\tLN:5674\r\n",
      "@SQ\tSN:chrUn_KI270511v1\tLN:8127\r\n",
      "@SQ\tSN:chrUn_KI270515v1\tLN:6361\r\n",
      "@SQ\tSN:chrUn_KI270507v1\tLN:5353\r\n",
      "@SQ\tSN:chrUn_KI270517v1\tLN:3253\r\n",
      "@SQ\tSN:chrUn_KI270529v1\tLN:1899\r\n",
      "@SQ\tSN:chrUn_KI270528v1\tLN:2983\r\n",
      "@SQ\tSN:chrUn_KI270530v1\tLN:2168\r\n",
      "@SQ\tSN:chrUn_KI270539v1\tLN:993\r\n",
      "@SQ\tSN:chrUn_KI270538v1\tLN:91309\r\n",
      "@SQ\tSN:chrUn_KI270544v1\tLN:1202\r\n",
      "@SQ\tSN:chrUn_KI270548v1\tLN:1599\r\n",
      "@SQ\tSN:chrUn_KI270583v1\tLN:1400\r\n",
      "@SQ\tSN:chrUn_KI270587v1\tLN:2969\r\n",
      "@SQ\tSN:chrUn_KI270580v1\tLN:1553\r\n",
      "@SQ\tSN:chrUn_KI270581v1\tLN:7046\r\n",
      "@SQ\tSN:chrUn_KI270579v1\tLN:31033\r\n",
      "@SQ\tSN:chrUn_KI270589v1\tLN:44474\r\n",
      "@SQ\tSN:chrUn_KI270590v1\tLN:4685\r\n",
      "@SQ\tSN:chrUn_KI270584v1\tLN:4513\r\n",
      "@SQ\tSN:chrUn_KI270582v1\tLN:6504\r\n",
      "@SQ\tSN:chrUn_KI270588v1\tLN:6158\r\n",
      "@SQ\tSN:chrUn_KI270593v1\tLN:3041\r\n",
      "@SQ\tSN:chrUn_KI270591v1\tLN:5796\r\n",
      "@SQ\tSN:chrUn_KI270330v1\tLN:1652\r\n",
      "@SQ\tSN:chrUn_KI270329v1\tLN:1040\r\n",
      "@SQ\tSN:chrUn_KI270334v1\tLN:1368\r\n",
      "@SQ\tSN:chrUn_KI270333v1\tLN:2699\r\n",
      "@SQ\tSN:chrUn_KI270335v1\tLN:1048\r\n",
      "@SQ\tSN:chrUn_KI270338v1\tLN:1428\r\n",
      "@SQ\tSN:chrUn_KI270340v1\tLN:1428\r\n",
      "@SQ\tSN:chrUn_KI270336v1\tLN:1026\r\n",
      "@SQ\tSN:chrUn_KI270337v1\tLN:1121\r\n",
      "@SQ\tSN:chrUn_KI270363v1\tLN:1803\r\n",
      "@SQ\tSN:chrUn_KI270364v1\tLN:2855\r\n",
      "@SQ\tSN:chrUn_KI270362v1\tLN:3530\r\n",
      "@SQ\tSN:chrUn_KI270366v1\tLN:8320\r\n",
      "@SQ\tSN:chrUn_KI270378v1\tLN:1048\r\n",
      "@SQ\tSN:chrUn_KI270379v1\tLN:1045\r\n",
      "@SQ\tSN:chrUn_KI270389v1\tLN:1298\r\n",
      "@SQ\tSN:chrUn_KI270390v1\tLN:2387\r\n",
      "@SQ\tSN:chrUn_KI270387v1\tLN:1537\r\n",
      "@SQ\tSN:chrUn_KI270395v1\tLN:1143\r\n",
      "@SQ\tSN:chrUn_KI270396v1\tLN:1880\r\n",
      "@SQ\tSN:chrUn_KI270388v1\tLN:1216\r\n",
      "@SQ\tSN:chrUn_KI270394v1\tLN:970\r\n",
      "@SQ\tSN:chrUn_KI270386v1\tLN:1788\r\n",
      "@SQ\tSN:chrUn_KI270391v1\tLN:1484\r\n",
      "@SQ\tSN:chrUn_KI270383v1\tLN:1750\r\n",
      "@SQ\tSN:chrUn_KI270393v1\tLN:1308\r\n",
      "@SQ\tSN:chrUn_KI270384v1\tLN:1658\r\n",
      "@SQ\tSN:chrUn_KI270392v1\tLN:971\r\n",
      "@SQ\tSN:chrUn_KI270381v1\tLN:1930\r\n",
      "@SQ\tSN:chrUn_KI270385v1\tLN:990\r\n",
      "@SQ\tSN:chrUn_KI270382v1\tLN:4215\r\n",
      "@SQ\tSN:chrUn_KI270376v1\tLN:1136\r\n",
      "@SQ\tSN:chrUn_KI270374v1\tLN:2656\r\n",
      "@SQ\tSN:chrUn_KI270372v1\tLN:1650\r\n",
      "@SQ\tSN:chrUn_KI270373v1\tLN:1451\r\n",
      "@SQ\tSN:chrUn_KI270375v1\tLN:2378\r\n",
      "@SQ\tSN:chrUn_KI270371v1\tLN:2805\r\n",
      "@SQ\tSN:chrUn_KI270448v1\tLN:7992\r\n",
      "@SQ\tSN:chrUn_KI270521v1\tLN:7642\r\n",
      "@SQ\tSN:chrUn_GL000195v1\tLN:182896\r\n",
      "@SQ\tSN:chrUn_GL000219v1\tLN:179198\r\n",
      "@SQ\tSN:chrUn_GL000220v1\tLN:161802\r\n",
      "@SQ\tSN:chrUn_GL000224v1\tLN:179693\r\n",
      "@SQ\tSN:chrUn_KI270741v1\tLN:157432\r\n",
      "@SQ\tSN:chrUn_GL000226v1\tLN:15008\r\n",
      "@SQ\tSN:chrUn_GL000213v1\tLN:164239\r\n",
      "@SQ\tSN:chrUn_KI270743v1\tLN:210658\r\n",
      "@SQ\tSN:chrUn_KI270744v1\tLN:168472\r\n",
      "@SQ\tSN:chrUn_KI270745v1\tLN:41891\r\n",
      "@SQ\tSN:chrUn_KI270746v1\tLN:66486\r\n",
      "@SQ\tSN:chrUn_KI270747v1\tLN:198735\r\n",
      "@SQ\tSN:chrUn_KI270748v1\tLN:93321\r\n",
      "@SQ\tSN:chrUn_KI270749v1\tLN:158759\r\n",
      "@SQ\tSN:chrUn_KI270750v1\tLN:148850\r\n",
      "@SQ\tSN:chrUn_KI270751v1\tLN:150742\r\n",
      "@SQ\tSN:chrUn_KI270752v1\tLN:27745\r\n",
      "@SQ\tSN:chrUn_KI270753v1\tLN:62944\r\n",
      "@SQ\tSN:chrUn_KI270754v1\tLN:40191\r\n",
      "@SQ\tSN:chrUn_KI270755v1\tLN:36723\r\n",
      "@SQ\tSN:chrUn_KI270756v1\tLN:79590\r\n",
      "@SQ\tSN:chrUn_KI270757v1\tLN:71251\r\n",
      "@SQ\tSN:chrUn_GL000214v1\tLN:137718\r\n",
      "@SQ\tSN:chrUn_KI270742v1\tLN:186739\r\n",
      "@SQ\tSN:chrUn_GL000216v2\tLN:176608\r\n",
      "@SQ\tSN:chrUn_GL000218v1\tLN:161147\r\n",
      "@SQ\tSN:chrX\tLN:156040895\r\n",
      "@SQ\tSN:chrY\tLN:57227415\r\n",
      "@SQ\tSN:chrY_KI270740v1_random\tLN:37240\r\n",
      "@PG\tID:minimap2\tPN:minimap2\tVN:2.26-r1175\tCL:minimap2 -c -a /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.mmi /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz\r\n",
      "@PG\tID:samtools\tPN:samtools\tPP:minimap2\tVN:1.17\tCL:samtools view -H /home/hice1/ynitta6/scratch/Project2/ReadAlignment/SRR099988_minimap.sam\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view -H ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9fa5cd74",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@SQ\tSN:chrY\tLN:57227415\r\n",
      "@SQ\tSN:chrY_KI270740v1_random\tLN:37240\r\n",
      "@PG\tID:minimap2\tPN:minimap2\tVN:2.26-r1175\tCL:minimap2 -c -a /home/hice1/ynitta6/scratch/Project2/ReadAlignment/hg38.mmi /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_1.Trimmed.fastq.gz /home/hice1/ynitta6/scratch/Project/data/Trimming/SRR099988_2.Trimmed.fastq.gz\r\n",
      "SRR099988.8890\t0\tchrX\t91436517\t4\t60M\t*\t0\t0\tAATGATGAAAAACTGAAGGAGGAATTTTCTTCCTTTGGGTCAATTAGTCNGNCNANANTG\tFFFFFF=FFCDDBDD;?EEEC;-@DFFF@EEEFFFEFDF3DD:BA####!#!#!#!#!##\tNM:i:5\tms:i:105\tAS:i:105\tnn:i:5\ttp:A:P\tcm:i:7\ts1:i:43\ts2:i:0\tde:f:0.0833\trl:i:0\r\n",
      "SRR099988.8891\t4\t*\t0\t0\t*\t*\t0\t0\tCACTCTTAACCTTCATTATATACTGGGGGGACCTCCCTGGTCTCTGTTGNTNCNANTNGA\tGCECGGGDFFEEFFEGGGGFGBGGEGFFED5FFFCEEBEE=EE?#####!#!#!#!#!##\trl:i:0\r\n",
      "SRR099988.8892\t4\t*\t0\t0\t*\t*\t0\t0\tTGGCCAACAGACAAGCAAGGATGCCTCACCTGCGGATGGTCCTGAGCAGNGNGNGNCNGG\tHHHHHHFHHHGHHHHHHHHGHHHHHGHCHGHHHHHFFHH=CEA;C####!#!#!#!#!##\trl:i:0\r\n",
      "SRR099988.8893\t0\tchr5\t150895682\t11\t60M\t*\t0\t0\tTTTTTCTCCTGTGTGAATTCTCTGATGTCCAATGAGATGTGACTTCTGGNANANGNCTTT\tHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHFFEHHEHHHAFABB@!#!#!#!####\tNM:i:4\tms:i:108\tAS:i:108\tnn:i:4\ttp:A:P\tcm:i:11\ts1:i:47\ts2:i:0\tde:f:0.0667\trl:i:0\r\n",
      "SRR099988.8894\t4\t*\t0\t0\t*\t*\t0\t0\tACCAAGGTGGAAGAGCGTCTCACCACCATCTGCCAGCGGGAAAACTCCTNCNANGNGGAC\tHHGHHGHFHHGHHHHHHHHEHHHHHHHHGHHGHHHFHFHD5FDB#####!#!#!#!####\trl:i:0\r\n",
      "SRR099988.8903\t4\t*\t0\t0\t*\t*\t0\t0\tCTGAAGAGTGGAAGTTGAAGTGGAAATATTGCTTGATGGAGAAACTATCNTNTNANCTGC\tHHGEFHFHDHECEE9E?AED@BD:A?ADADBCCFA4A5D9>=BA@####!#!#!#!####\trl:i:0\r\n",
      "SRR099988.8916\t4\t*\t0\t0\t*\t*\t0\t0\tCCCTACAGAAAATATGATTGGATCTCGGGACTTGAGGATTTTCTTATACNANCNANGNTT\tHHHHHHHHHHHHHHGH@HHHHFGHHHHHHDHHHHEFHFEHFBB:H288?!#!#!#!#!##\trl:i:0\r\n",
      "SRR099988.8917\t4\t*\t0\t0\t*\t*\t0\t0\tTTCCTGTGATAGTGGTGGAGAGTGTGCCTCGATAACTCTGTCCATCACCNCNGNANCAGT\tHHHHHHHHHHGHFHHHHHHFFHDFDHFHBHF:BFEEEFEFEHE?#####!#!#!#!####\trl:i:0\r\n",
      "SRR099988.8918\t4\t*\t0\t0\t*\t*\t0\t0\tGCTGCTTTGGTTCTGGAACAATCTTCAGGAAGCTGTTCGTTCGGGAGTANANGNGTGGGG\tHHHHHHHHHHHHHHHHHGHFHHHHHHHHGHHHHHHDFHHFHHHFH####!#!#!######\trl:i:0\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 466 ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam |tail -n -12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3af73d7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "We ran !samtools view -H command on our sam file to view the header and header line. In the header line, there is\n",
    "a section called \"SO,\" which stands for Sort Order. You can check \"SO\" to see if your sam file is sorted or not.\n",
    "Looking at \"SO\" in HD(header definition) of our sam file, it says \"unsorted.\" Therefore, this sam file is not sorted.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f05ae2be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 14 files and 1 in-memory blocks...\r\n"
     ]
    }
   ],
   "source": [
    "!samtools sort -o ~/scratch/Project2/ReadAlignment/SRR099988_sorted_minimap.sam ~/scratch/Project2/ReadAlignment/SRR099988_minimap.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b5a61e2c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@HD\tVN:1.6\tSO:coordinate\r\n",
      "@SQ\tSN:chr1\tLN:248956422\r\n",
      "@SQ\tSN:chr10\tLN:133797422\r\n",
      "@SQ\tSN:chr11\tLN:135086622\r\n",
      "@SQ\tSN:chr11_KI270721v1_random\tLN:100316\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 5 ~/scratch/Project2/ReadAlignment/SRR099988_sorted_minimap.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "da809aa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ~/scratch/Project2/ReadAlignment/SAM_BAM_files\n",
    "!mv ~/scratch/Project2/ReadAlignment/SRR099988_sorted_minimap.sam ~/scratch/Project2/ReadAlignment/SAM_BAM_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "57b7503c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!samtools view -bS ~/scratch/Project2/ReadAlignment/SAM_BAM_files/SRR099988_sorted_minimap.sam -o ~/scratch/Project2/ReadAlignment/SAM_BAM_files/SRR099988.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6537a5e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 40G\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson 5.7G Oct 30 23:21 SRR099988.bam\r\n",
      "-rw-r--r-- 1 ynitta6 gtperson  35G Oct 30 21:55 SRR099988_sorted_minimap.sam\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh ~/scratch/Project2/ReadAlignment/SAM_BAM_files/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "51df61f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "176131008\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view -c ~/scratch/Project2/ReadAlignment/SAM_BAM_files/SRR099988_sorted_minimap.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e76ac5f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "In our sam file, there are 176131008 raw sequences.\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e6efcb2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "168002305\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view -c -F 4 ~/scratch/Project2/ReadAlignment/SAM_BAM_files/SRR099988_sorted_minimap.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "842a4cdd",
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "In our sam file, there are 168002305 mapped reads.\n",
    "'''"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
